src/com/google/javascript/jscomp/CodeGenerator.java

```diff
@@ -436,7 +436,15 @@ protected void add(Node node, Context context) {
       case ITER_SPREAD:
       case OBJECT_SPREAD:
         add("...");
-        add(node.getFirstChild());
+        if (first.getIsParenthesized()) {
```

Comment:
I believe the right way to do this is actually to use

`addExpr(first, NodeUtil.precedence(type), Context.OTHER);`

instead of this if-statement.

Reply:
I figured as much, since this seemed like a common problem for a code generator, I just couldn't find the correct way.
I've force pushed the branch with the fix.